src/torrent.cpp

```diff
@@ -8332,17 +8332,17 @@ namespace libtorrent
 
 	namespace {
 
-	int clamped_subtract(int a, int b)
+	std::uint16_t clamped_subtract_u16(int a, int b)
 	{
 		if (a < b) return 0;
-		return a - b;
+		return std::uint16_t(a - b);
 	}
 
-	int clamped_subtract_s16(int a, int b)
+	std::int16_t clamped_subtract_s16(int a, int b)
 	{
 		if (a + (std::numeric_limits<std::int16_t>::min)() < b)
```

Comment:
I don't think the test is needed in `clamped_subtract_u16` since the the min unsigned is `0`. Could you confirm this?